core/controllers/suggestion.py

```diff
@@ -344,8 +352,24 @@ def get(self, target_type, suggestion_type):
         """
         self._require_valid_suggestion_and_target_types(
             target_type, suggestion_type)
+        topic_name = self.request.get('topic_name', None)
+
+        opportunity_summary_exp_ids_specific_to_topic = []
+        if topic_name is not None:
+            topic = topic_fetchers.get_topic_by_name(topic_name)
+            if topic is not None:
+                exploration_opportunity_summaries = (
+                    opportunity_services.
+                    get_exploration_opportunity_summaries_by_topic_id(
+                    topic.id))
+
+                opportunity_summary_exp_ids_specific_to_topic = [
+                    opportunity.id for opportunity
+                    in exploration_opportunity_summaries]
+
         suggestions = suggestion_services.get_reviewable_suggestions(
-            self.user_id, suggestion_type)
+            self.user_id, suggestion_type,
```

Comment:
I don't understand why we are passing both opportunity_summary_exp_ids_specific_to_topic and topic_name. Isn't the former enough? The latter only seems to be used as a boolean.

Note that the case of a topic with no exp IDs is represented by [], and the case of "no topic passed at all" is represented by None.

Reply:
opportunity_summary_exp_ids_specific_to_topic can be an empty list in two cases: 
1. When a topic name was passed, but no exp ID's exists, and in this case, we want to return 0 suggestions. 
2. When no topic name was passed as part of the request, and in this case, we want to return all suggestions. 

In order to differentiate between the two, we are passing topic_name.

Reply:
See the second paragraph in my previous comment, which provides another way to differentiate the two (which I think is also semantically correct).

Reply:
Oh, just clarifying, when you say "and the case of "no topic passed at all" is represented by None.", you are referring to having the value of "opportunity_summary_exp_ids_specific_to_topic" as None, correct?

Reply:
Yup.